pulsar-functions/windowing/src/main/java/org/apache/pulsar/functions/windowing/WindowContextImpl.java

```diff
@@ -117,7 +117,10 @@ public ByteBuffer getState(String key) {
 
     @Override
     public String getUserConfigValue(String key) {
-        return this.getUserConfigValue(key);
+        if (this.context.getUserConfigValue(key).isPresent()) {
+            return this.context.getUserConfigValue(key).get().toString();
```

Comment:
This API is actually incorrect. @cckellogg can you change the signature from returning a String to returning an Optional<Object> as this is the case for functions:

https://github.com/apache/pulsar/blob/master/pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java#L238

Reply:
@jerrypeng this uses a different interface https://github.com/apache/pulsar/blob/master/pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/WindowContext.java
maybe the can be consolidated in another pr?